### PR TITLE
Upgrade to Camel Quarkus 3.0.0-M2

### PR DESCRIPTION
**Pull request by ppalaga**
WiP

Camel Quarkus 3.0.0-M2 staging repo needs to get removed before merging

Ideally, https://github.com/quarkusio/quarkus-platform/pull/834 should be merged first and this one rebased.

### CODE REVIEW

**Comment by ppalaga** [comment]
d82ef86: rebased

**Comment by jamesnetherton** [comment]
The master failure is due to this change:

https://github.com/apache/camel-quarkus/commit/2a797547c15839b718531b81ad1512cce1bc2f6d

It was one of the tests that we never took to the platform, because it wants to start a secondary (native) process to test failover etc.

Presumably we can temporarily exclude it in the platform for now...

**Comment by aldettinger** [comment]
I was not able to push the fix directly to @ppalaga branch, so finally end up with a [pr](https://github.com/ppalaga/quarkus-platform/pull/1)

**Comment by ppalaga** [comment]
> The master failure is due to this change:
> 
> [apache/camel-quarkus@2a79754](https://github.com/apache/camel-quarkus/commit/2a797547c15839b718531b81ad1512cce1bc2f6d)

Ups, sorry for breaking it & thanks for inestigating, @jamesnetherton !

**Comment by ppalaga** [comment]
3d0c1a5:

* Rebased
* Excluded the failing master test that should not be there at all 
* Excluded the failing tests in quarkus-cxf

Should be good to merge now let's see whether the CI is going to pass.

**Comment by ppalaga** [comment]
Good, the CI passed. @gsmet I hope we are on a good way to catch the 3.1.0 train?

File: .gitignore [hunk 1]
```diff
@@ -36,4 +36,16 @@ build.log
...
+# Temporary data ignored in quarkus-cxf tests
+/generated-platform-project/quarkus-cxf/integration-tests/quarkus-cxf-integration-test-ws-rm/derby.log
+
+# Workaround for https://github.com/quarkusio/quarkus/issues/33305
+/generated-platform-project/quarkus-cxf/integration-tests/Java2wsTest/
+/generated-platform-project/quarkus-camel/integration-tests/Java2wsTest/
+
+# Workaround for https://github.com/quarkiverse/quarkus-cxf/issues/900
+/generated-platform-project/quarkus-cxf/integration-tests/quarkus-cxf-integration-test-logging/src/main/resources/wsdl/CalculatorService.wsdl
+/generated-platform-project/quarkus-cxf/integration-tests/quarkus-cxf-integration-test-ws-security-client/src/main/resources/wsdl/dir/WssCalculatorService.wsdl
```
Comment (gsmet) [inline .gitignore:51]: Why do you generate temporary data in the source files? It looks like a bad practice to me.

(But thanks for fixing this as it was a bit annoying and I was wondering if I was doing something wrong)

**Review by gsmet — APPROVED** [review]

**Pull request merged**